### DISCUSSION TITLE
Features brainstorm

### DISCUSSION THREAD

**Discussion by mslaursen** [SD-2-semester/DLS-semester-project#2] [Ideas]
Can be functional or non-functional

**Comment by mslaursen** [comment]
Users
Servers
Channels
Group chat
Private chat
Status (away, online, etc.)
Notifications
Streaming(?)
Voice chat (?)

If we have extra time and create voice chat we could have this unique feature that discord does not have:
A special 1 man admin voice channel, that can broadcast his voice to every user connected to the voicechat service.

**Comment by Moshizzl3** [comment]
```mermaid
graph LR
    Users -->|authenticate| Auth
    Users -->|join| Servers
    Users -->|update| Status
    Servers -->|create/manage| Channels
    Channels -->|host| GroupChat
    Channels -->|host| PrivateChat
    GroupChat -->|send/receive messages| Notifications
    PrivateChat -->|send/receive messages| Notifications
    Status -->|update status| Notifications
    Notifications -->|notify users| Users
```
